src/main/java/com/labsynch/labseer/service/ParentSwapStructuresServiceImpl.java

```diff
@@ -0,0 +1,91 @@
+package com.labsynch.labseer.service;
+
+import java.util.Collection;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.labsynch.labseer.domain.Parent;
+import com.labsynch.labseer.dto.ParentDTO;
+import com.labsynch.labseer.dto.ParentSwapStructuresDTO;
+import com.labsynch.labseer.dto.ParentValidationDTO;
+
+@Service
+public class ParentSwapStructuresServiceImpl implements ParentSwapStructuresService {
+
+    Logger logger = LoggerFactory.getLogger(ParentSwapStructuresServiceImpl.class);
+
+    @Autowired
+    public ParentService parentService;
+
+    @Override
+    public boolean swapParentStructures(ParentSwapStructuresDTO parentSwapStructuresDTO) {
+        String corpName1 = parentSwapStructuresDTO.getCorpName1();
+		String corpName2 = parentSwapStructuresDTO.getCorpName2();
+
+		Parent parent1;
+		Parent parent2;
+		try {
+			parent1 = Parent.findParentsByCorpNameEquals(corpName1).getSingleResult();
+			parent2 = Parent.findParentsByCorpNameEquals(corpName2).getSingleResult();
+		} catch (Exception e) {
+			logger.error("Caught error while fetching parent", e);
+            return false;
+		}
+
+		ParentValidationDTO validationDTO1;
+		ParentValidationDTO validationDTO2;
+		try {
+			validationDTO1 = parentService.validateUniqueParent(parent1);
+			validationDTO2 = parentService.validateUniqueParent(parent2);
```

Comment:
I lack knowledge of this but is `validateUniqueParent` needed if it was instantiated by `findParentsByCorpNameEquals` above

Reply:
If the item exists in the db then it should already be valid and here the logic is merely sanity check. I am open to removing the check from here entirely.

Reply:
I am keeping the checks for now!